src/fhirtypes/common.ts

```diff
@@ -0,0 +1,75 @@
+import { ResolveFn, StructureDefinition, PathPart, ElementDefinition } from '.';
+
+/**
+ * This function sets an instance property of an SD or ED if possible
+ * @param {StructureDefinition | ElementDefinition} - The instance to fix a value on
+ * @param {string} path - The path to fix a value at
+ * @param {any} value - The value to fix
+ * @param {ResolveFn} resolve - A function that can resolve a type to a StructureDefinition instance
+ */
+function setPropertyOnInstance(
+  instance: StructureDefinition | ElementDefinition,
+  path: string,
+  value: any,
+  resolve: ResolveFn = () => undefined
+): void {
+  const structureDefinitionStructureDefinition = instance.getOwnStructureDefinition(resolve);
+  const { fixedValue, pathParts } = structureDefinitionStructureDefinition.validateValueAtPath(
+    path,
+    value,
+    resolve
+  );
+  if (fixedValue != null) {
+    // If we can fix the value on the StructureDefinition StructureDefinition, then we can set the
+    // instance property here
+    let current: any = instance;
+    for (const [i, pathPart] of pathParts.entries()) {
+      const key = pathPart.base;
+      // If this part of the path indexes into an array, the index will be the last bracket
+      const index = getArrayIndex(pathPart);
+      if (index != null) {
+        // If the array doesn't exist, create it
+        if (current[key] == null) current[key] = [];
+        // If the index doesn't exist in the array, add it and lesser indices
+        let j = current[key].length;
+        for (; j < index; j++) {
+          current[key].push(undefined);
+        }
+        if (j === index) {
+          current[key].push({});
+        }
```

Comment:
I don't think it is necessary to use this loop here. If you assign a value to an array index beyond its current length, the elements between the current end and the new element will automatically be set to `undefined`. It should be fine to do something like: `if(index >= current[key].length) { current[key][index] = {}; }`

Reply:
Oh cool! I did not know that. That's so much neater.

Reply:
Woah.  I did not know that either!  Thanks for the tip, @mint-thompson!